src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```
@@ -137,6 +137,9 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 {
                     await RestoreAsync(request.ForceRestore, request.RestoreSource, token);
                 }
+                catch (OperationCanceledException) when (token.IsCancellationRequested)
```

Why do you swallow `OperationCanceledException` when cancellation is requested? The contract is that if a token is canceled but the method completes without throwing that exception, it must have completed the operation. If the user cancels and you abort without completing, you should throw the exception so your caller knows it was not completed.

---

If your pattern is this method returns `false` in cancellation events, I'd count on that being documented in xml doc comments on the method.

---

This change to swallow `OperationCancelledException` is simply to avoid polluting activity log with exception stacks when cancellation is requested. There's no other change in behavior rather than that nuance.

---

OK, if this isn't a public API that outsiders can call, I guess that makes sense. But this doesn't appear to be the top-level method. So IMO the new catch should be in the top-level caller rather than this method, that way you can still suppress the activitylog entry, but maintain the typical contract of throwing when cancelling from your library-level method.